codec/reflectcodec/type_codec.go

```diff
@@ -520,6 +561,36 @@ func (c *genericCodec) unmarshal(p *wrappers.Packer, value reflect.Value, maxSli
 		// Assign to the top-level struct's member
 		value.Set(v)
 		return nil
+	case reflect.Map:
+		numElts32 := p.UnpackInt()
+		if numElts32 > c.maxSliceLen || numElts32%2 != 0 {
+			return fmt.Errorf("%w; array length, %d, exceeds maximum length, %d",
+				codec.ErrMaxSliceLenExceeded,
+				numElts32,
+				c.maxSliceLen,
+			)
+		}
+
+		numElts := int(numElts32 / 2)
+		value.Set(reflect.MakeMapWithSize(value.Type(), numElts))
+		keyType := value.Type().Key()
+		valueType := value.Type().Elem()
+
+		for i := 0; i < numElts; i++ {
+			keyValue := reflect.New(keyType).Elem()
+			valueValue := reflect.New(valueType).Elem()
+
+			if err := c.unmarshal(p, keyValue, c.maxSliceLen); err != nil {
+				return fmt.Errorf("couldn't unmarshal map key: %w", err)
```

Comment:
Maybe we should include the index in the error messages to make debugging easier